File: backend/OlivePlatform/OlivePlatform.Application/Features/OlivePurchases/Responses/OlivePurchaseForListResponse.cs
using OlivePlatform.Domain.Enums;

namespace OlivePlatform.Application.Features.OlivePurchases.Responses;

public class OlivePurchaseForListResponse
{
    public int Total { get; set; }

    public int Id { get; set; }

    public string Reference { get; set; } = null!;

    public string SupplierName { get; set; } = null!;

    public DateOnly PurchaseDate { get; set; }

    public PurchaseStatus Status { get; set; }

    public DateTime CreatedAt { get; set; }

    public decimal TotalQuantityKg { get; set; }
}

File: frontend/olive-platform/src/features/olivePurchases/data/responses/OlivePurchaseForListResponse.ts
import type { ProductionStatus } from "../../../production/domain/entities/ProductionStatus";
import type { PurchaseStatus } from "../../domain/entities/PurchaseStatus";

export type OlivePurchaseForListResponse = {
  id: number;
  reference: string;
  supplierName: string;
  purchaseDate: string;
  status: PurchaseStatus;
  createdAt: string;
  pressed: ProductionStatus;
  analyseStatus: ProductionStatus;
  totalQuantityKg: number;
  canLaunchPression: boolean;
};